contrib/exporters/vpclogs/mod/object_header.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/logging"
+)
+
+const version = "0.0.1"
+
+type objectHeaderVpc struct {
+	keyValues map[string]string
+}
+
+// AddObjHeader inserts the object global paramateres and wraps the flows
+func (h *objectHeaderVpc) AddObjHeader(flows []interface{}, startTime string, endTime string) interface{} {
+	var augmentedObject map[string]interface{}
+	augmentedObject = make(map[string]interface{})
+	for key, value := range h.keyValues {
+		augmentedObject[key] = value
+	}
+	augmentedObject["version"] = version
+	augmentedObject["capture_start_time"] = startTime
+	augmentedObject["capture_end_time"] = endTime
+	augmentedObject["number_of_flow_logs"] = len(flows)
+	augmentedObject["flow_logs"] = flows
+	// TBD can we do better with state?
+	augmentedObject["state"] = "skip data"
+
```

Comment:
`collector_crn`, `attached_endpoint_type`, and `vpc_crn` are non-optional fields that are not present here
`network_interface_id` and `instance_crn` are optional fields that are not present

Will these be added later, or will we request these be dropped?

Reply:
These various IDs are not known to Skydive, so they must be provided by the higher-level management plane. These should be provided in the vpclogs.yml file and are incorporated into the flow logs via the keyValues in the objectHeaderVpc structure. They are then copied directly in to the header of the flow object.